Tests/BSGConfigurationBuilderTests.m

```diff
@@ -0,0 +1,494 @@
+    XCTAssertNil(config);
+}
+
+- (void)testDecodeAutoCaptureSessions {
+    BugsnagConfiguration *config =
+        [BSGConfigurationBuilder configurationFromOptions:@{
+            @"autoSessions" : @NO,
+            @"apiKey" : @"2343a136bff"
+        }];
+    XCTAssertNotNil(config);
+    XCTAssertFalse(config.shouldAutoCaptureSessions);
+    XCTAssertEqualObjects(@"2343a136bff", config.apiKey);
+    config = [BSGConfigurationBuilder configurationFromOptions:@{
+        @"autoSessions" : @YES,
+        @"apiKey" : @"2343a136bff"
+    }];
+    XCTAssertNotNil(config);
+    XCTAssertTrue(config.shouldAutoCaptureSessions);
+    XCTAssertEqualObjects(@"2343a136bff", config.apiKey);
+}
+
+- (void)testDecodeAutoCollectBreadcrumbs {
+    BugsnagConfiguration *config =
+        [BSGConfigurationBuilder configurationFromOptions:@{
+            @"autoBreadcrumbs" : @NO,
+            @"apiKey" : @"1343a136bff"
+        }];
+    XCTAssertNotNil(config);
+    XCTAssertFalse(config.automaticallyCollectBreadcrumbs);
+    XCTAssertEqualObjects(@"1343a136bff", config.apiKey);
+    config = [BSGConfigurationBuilder configurationFromOptions:@{
+        @"autoBreadcrumbs" : @YES,
+        @"apiKey" : @"1343a136bff"
+    }];
+    XCTAssertNotNil(config);
+    XCTAssertTrue(config.automaticallyCollectBreadcrumbs);
+    XCTAssertEqualObjects(@"1343a136bff", config.apiKey);
+}
+
+- (void)testDecodeAutoCollectBreadcrumbsInvalidType {
+    BugsnagConfiguration *config =
+        [BSGConfigurationBuilder configurationFromOptions:@{
+            @"autoBreadcrumbs" : @"foo",
+            @"apiKey" : @"5628acc"
+        }];
+    XCTAssertNil(config);
+}
+
+- (void)testDecodeAutoCollectBreadcrumbsWithoutApiKey {
+    BugsnagConfiguration *config = [BSGConfigurationBuilder
+        configurationFromOptions:@{@"autoBreadcrumbs" : @NO}];
+    XCTAssertNil(config);
+    config = [BSGConfigurationBuilder
+        configurationFromOptions:@{@"autoBreadcrumbs" : @YES}];
+    XCTAssertNil(config);
+}
+
+- (void)testDecodeEndpointsInvalidTypes {
+    BugsnagConfiguration *config = [BSGConfigurationBuilder configurationFromOptions:@{
+        @"endpoints" : @"foo",
+        @"apiKey" : @"5628acc"
+    }];
+    XCTAssertNil(config);
+    config = [BSGConfigurationBuilder configurationFromOptions:@{
+        @"endpoints" : @[ @"http://example.com", @"http://foo.example.com" ],
+        @"apiKey" : @"5628acc"
+    }];
+    XCTAssertNil(config);
+    config = [BSGConfigurationBuilder configurationFromOptions:@{
+        @"endpoints" : @{},
+        @"apiKey" : @"5628acc"
+    }];
+    XCTAssertNil(config);
+}
+
+- (void)testDecodeEndpointsUnknownKeys {
+    BugsnagConfiguration *config = [BSGConfigurationBuilder configurationFromOptions:@{
+        @"endpoints" : @{
+            @"notify" : @"https://notify.example.com",
+            @"sessions" : @"https://sessions.example.com",
+            @"florps" : @"https://florps.example.com",
```

Comment:
We should just ignore invalid keys, as long as any required information (i.e. notify + sessions) is there